src/core/tsi/ssl_transport_security_utils.cc

```diff
@@ -375,4 +380,79 @@ absl::StatusOr<std::string> AkidFromCrl(X509_CRL* crl) {
   return ret;
 }
 
+bool IsPemCertificateChainNonEmptyAndValid(absl::string_view cert_chain_pem) {
+  if (cert_chain_pem.empty()) {
+    return false;
+  }
+  BIO* in = BIO_new_mem_buf(cert_chain_pem.data(), cert_chain_pem.size());
+  if (in == nullptr) {
+    return false;
+  }
+  uint8_t* cert_data = nullptr;
+  long cert_length = 0;
+  size_t number_of_valid_certs = 0;
+  while (PEM_bytes_read_bio(&cert_data, &cert_length, /*pnm=*/nullptr,
+                            PEM_STRING_X509, in, /*cb=*/nullptr,
+                            /*u=*/nullptr)) {
+    // Try to form the X.509 object from the DER-encoding.
+    const uint8_t* der = cert_data;
+    X509* x509 = d2i_X509(/*out=*/nullptr, &der, cert_length);
+    if (x509 == nullptr) {
```

Comment:
Note that, by not checking `der` against `cert_data + cert_length`, you're also accepting PEM blocks with extra trailing data. Although the functions inside OpenSSL and BoringSSL also accept that due to a bug... which we hope to fix in the future.

Reply:
Ah great call! Is this still relevant now that we switched to using `PEM_read_bio_X509`?